File: android/app/src/main/java/com/example/cfsamplewebviewfl/WebViewFactory.java
package com.example.cfsamplewebviewfl;

import android.annotation.SuppressLint;
import android.content.Context;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import java.util.Map;

import io.flutter.plugin.common.StandardMessageCodec;
import io.flutter.plugin.platform.PlatformView;
import io.flutter.plugin.platform.PlatformViewFactory;

public class WebViewFactory extends PlatformViewFactory {
    public WebView webView;
    private final CfJsBridge cfJsBridge;

    public WebViewFactory(Context context) {
        super(StandardMessageCodec.INSTANCE);
        this.cfJsBridge = new CfJsBridge(context);
    }

    @Override
    public PlatformView create(Context context, int viewId, Object args) {
        webView = initAndGetWebView(context, args);
        return new CustomWebViewPlatform(webView);
    }

    @SuppressLint("SetJavaScriptEnabled")
    private WebView initAndGetWebView(Context context, Object args) {
        WebView webView = new WebView(context);
        webView.getSettings().setJavaScriptEnabled(true);
        webView.setWebViewClient(new WebViewClient());
        webView.addJavascriptInterface(cfJsBridge, "Android");

        if (args instanceof Map) {
            Map<?, ?> params = (Map<?, ?>) args;
            Object webUrl = params.get("webUrl");
            if (webUrl != null) {
                webView.loadUrl(webUrl.toString());
            }
        }
        return webView;
    }
}

File: android/app/src/main/kotlin/com/example/cfsamplewebviewfl/WebViewFactory.kt
package com.example.cfsamplewebviewfl

import android.annotation.SuppressLint
import android.content.Context
import android.view.View
import android.webkit.WebView
import android.webkit.WebViewClient
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

internal class WebViewFactory(private val context: Context) :
    PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    var webView: WebView? = null
    private val cfJsBridge: CFJsBridge by lazy { CFJsBridge(context) }
    override fun create(context: Context?, viewId: Int, args: Any?): PlatformView {
        webView = initAndGetWebView(context, args)
        return CustomWebViewPlatform(webView)
    }

    @SuppressLint("SetJavaScriptEnabled")
    private fun initAndGetWebView(context: Context?, args: Any?): WebView {
        val webView = WebView(context!!)
        webView.settings.javaScriptEnabled = true
        webView.webViewClient = WebViewClient()
        webView.addJavascriptInterface(cfJsBridge, "Android")
        val params = args as Map<*, *>
        val webUrl = params["webUrl"]
        if (webUrl != null) {
            webView.loadUrl(webUrl.toString())
        }
        return webView
    }
}

internal class CustomWebViewPlatform(private val webView: WebView?) : PlatformView {
    override fun getView(): View? = webView
    override fun dispose() {
        webView?.destroy()
    }
}